### DISCUSSION TITLE
How to import utils patterns for TS?

### DISCUSSION THREAD

**Discussion by qertis** [quasarframework/quasar#16091] [General - Components / Directives / etc]
TypeScript + ESLint Error  `Unsafe member access .testPattern on an `any` value`

```ts
// ...
import patterns from 'quasar/src/utils/patterns'

const consumerValid = computed(() => {
  return patterns.testPattern.email(email.value)
})
```

![image](https://github.com/quasarframework/quasar/assets/3893228/3789196f-8c0c-41cc-9828-c1d0f084c4ba)

**Comment by yusufkandemir** [comment]
It was an undocumented feature, that's why it doesn't have type definitions. It will be documented and will have typings with #16092, but you should use the direct import from `quasar`, not a deep import like you are doing.
